examples/api/lib/widgets/editable_text/editable_text.on_content_inserted.0.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Flutter code sample for EditableText.onContentInserted
+
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const KeyboardInsertedContentApp());
+
+class KeyboardInsertedContentApp extends StatelessWidget {
+  const KeyboardInsertedContentApp({super.key});
+
+  static const String _title = 'Keyboard Inserted Content Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: _title,
+      home: KeyboardInsertedContentDemo(),
+    );
+  }
+}
+
+class KeyboardInsertedContentDemo extends StatefulWidget {
+  const KeyboardInsertedContentDemo({super.key});
+
+  @override
+  State<KeyboardInsertedContentDemo> createState() => _KeyboardInsertedContentDemoState();
+}
+
+class _KeyboardInsertedContentDemoState extends State<KeyboardInsertedContentDemo> {
+  final TextEditingController _controller = TextEditingController();
+  Uint8List? bytes;
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Keyboard Inserted Content Sample')),
+      body: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          const Text("Here's a text field that supports inserting only png or gif content:"),
+          TextField(
+            controller: _controller,
+            contentInsertionMimeTypes: const <String>['image/png', 'image/gif'],
+            onContentInserted: (KeyboardInsertedContent data) async {
+              if (data.data != null) {
+                setState(() {
+                  bytes = data.data;
+                });
+              }
+            },
+          ),
+          if (bytes != null)
+            const Text("Here's the most recently inserted content:"),
+          if (bytes != null)
+            Image.memory(bytes!),
+        ],
+      ),
+    );
+  }
+}
```

Comment:
Can you add a test for this example?  I keep forgetting but we're trying to make sure all of these examples have tests too.  Here's a simple example: https://github.com/flutter/flutter/blob/master/examples/api/test/material/switch/switch.0_test.dart

I think you can do something similar to what you did in your editable_text_test.dart test to send some fake data to onContentInserted.  Then expect that the Image is shown.

Reply:
I talked about this in your previous review, this is what's going on:

I'm not able to get this going... it seems that the input client is either ignoring or not getting notified about the mock commit content invocation?

This is the code:

```dart
testWidgets('Image.memory displays committed content', (WidgetTester tester) async {
    await tester.pumpWidget(
      const example.ContentCommittedApp(),
    );

    expect(find.text('Content Committed Sample'), findsOneWidget);

    const String uri = 'content://com.google.android.inputmethod.latin.fileprovider/test.png';
    const List<int> kBlueSquarePng = <int>[
      0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a, 0x00, 0x00, 0x00, 0x0d, 0x49,
      0x48, 0x44, 0x52, 0x00, 0x00, 0x00, 0x32, 0x00, 0x00, 0x00, 0x32, 0x08, 0x06,
      0x00, 0x00, 0x00, 0x1e, 0x3f, 0x88, 0xb1, 0x00, 0x00, 0x00, 0x48, 0x49, 0x44,
      0x41, 0x54, 0x78, 0xda, 0xed, 0xcf, 0x31, 0x0d, 0x00, 0x30, 0x08, 0x00, 0xb0,
      0x61, 0x63, 0x2f, 0xfe, 0x2d, 0x61, 0x05, 0x34, 0xf0, 0x92, 0xd6, 0x41, 0x23,
      0x7f, 0xf5, 0x3b, 0x20, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44,
      0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44,
      0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44, 0x44,
      0x44, 0x44, 0x44, 0x36, 0x06, 0x03, 0x6e, 0x69, 0x47, 0x12, 0x8e, 0xea, 0xaa,
      0x00, 0x00, 0x00, 0x00, 0x49, 0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82,
    ];
    final ByteData? messageBytes = const JSONMessageCodec().encodeMessage(<String, dynamic>{
      'args': <dynamic>[
        -1,
        'TextInputAction.commitContent',
        jsonDecode('{"mimeType": "image/png", "data": $kBlueSquarePng, "uri": "$uri"}'),
      ],
      'method': 'TextInputClient.performAction',
    });

    try {
      await ServicesBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
        'flutter/textinput',
        messageBytes,
            (ByteData? _) {},
      );
    } catch (_) {}

    await tester.pump();
    final Size imgSize = tester.getSize(find.byType(Image));
    expect(imgSize.width.roundToDouble(), 50.0);
    expect(imgSize.height.roundToDouble(), 50.0);
  });
```

The `onContentCommitted` callback of the text field is never executed.

Reply:
Sorry I missed the discussion above about this test.

I would expect this to work since your test in editable_text_test.dart works.  Is it because you haven't interacted with the field first to open the input connection like you did in that test?

```dart
    await tester.tap(find.byType(EditableText));
    await tester.enterText(find.byType(EditableText), 'test');
    await tester.idle();
```

Reply:
Looks like you got the test working 👍